### PR TITLE
fix cache persistent default

### LINKED ISSUE #119: Breaking metadatastorage change in quay.io/cephcsi/cephfsplugin:v0.3.0
A change has been made in `master` concerning the following files:

deploy/cephfs/kubernetes/csi-cephfsplugin.yaml: 52
`     - "--metadatastorage=k8s_configmap"`

pkg/util/cachepersister.go: 38:52
```
    if metadataStore == "k8s_configmap" {
		glog.Infof("cache-perister: using kubernetes configmap as metadata cache persister")
		k8scm := &K8sCMCache{}
		k8scm.Client = NewK8sClient()
		k8scm.Namespace = GetK8sNamespace()
		return k8scm, nil
	} else if metadataStore == "node" {
		glog.Infof("cache-persister: using node as metadata cache persister")
		nc := &NodeCache{}
		nc.BasePath = PluginFolder + "/" + driverName
		return nc, nil
	}
    return nil, errors.New("cache-persister: couldn't parse metadatastorage flag")
```
    
`cachepersister.go` now explicitly requires the --metadatastorage flag to be passed in.

However, the Docker image build has been updated with the new code, but the tag has *remained* v0.3.0. This means that an existing deployment of the csi-cephfsplugin DaemonSet, which does not have the `"--metadatastorage=k8s_configmap"` argument in `csi-cephfsplugin.yaml`, will begin to break as new nodes are added to the k8s cluster and the new nodes download the latest copy of the Docker image.

This breaking change should be tagged as either v0.3.1 or v0.4.0, depending on your philosophy of SemVer

### PR DESCRIPTION
**Pull request by rootfs**
fix #119 
cc @mickymiek

### CODE REVIEW

**Comment by mickymiek** [comment]
Yep, haven't thought about this issue. Seems better to default cachepersister to node.

File: rbd/main.go [hunk 1]
```diff
@@ -35,7 +35,7 @@ var (
 	driverName      = flag.String("drivername", "csi-rbdplugin", "name of the driver")
 	nodeID          = flag.String("nodeid", "", "node id")
 	containerized   = flag.Bool("containerized", true, "whether run as containerized")
-	metadataStorage = flag.String("metadatastorage", "", "metadata persistence method [node|k8s_configmap]")
+	metadataStorage = flag.String("metadatastorage", "node", "metadata persistence method [node|k8s_configmap]")
```
Comment (gman0) [inline rbd/main.go]: For consistency, this should then also go in the cephfs driver

File: rbd/main.go [hunk 2]
```diff
@@ -50,6 +50,10 @@ func main() {
 		os.Exit(1)
 	}
 
+	if *metadataStorage != "node" && *metadataStorage != "k8s_configmap" {
```
Comment (gman0) [inline rbd/main.go]: The validation is already done here: 
https://github.com/ceph/ceph-csi/blob/master/pkg/util/cachepersister.go#L38-L50

**Comment by UrbanWizardry** [comment]
I realise I'm late to this as it's already been merged, but what it the logic behind choosing "node" for the default in code, but leaving the command-line arg in `deploy/cephfs/kubernetes/csi-cephfsplugin.yaml` as ` k8s_configmap` ?

**Comment by gman0** [comment]
We don't want to have k8s-specific defaults in the code, but it's ok to default to configmaps for k8s deployments. Also, this way it will have consistent behavior with older versions.

**Comment by UrbanWizardry** [comment]
Makes perfect sense, thanks for the fast response.

**Pull request merged**